File: Java/src/test/java/at/nagarro/exercises/bank/AccountServiceTest.java
package at.nagarro.exercises.bank;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class AccountServiceTest {

    @Test
    void shouldContainABalanceWithTheSumOfAllTransactions() {
        var owner = new Customer();
        owner.firstName = "Dagobert";
        owner.surName = "Duck";
        var account = new Account(owner);

        account.deposit(new Money(80.7));
        account.withdraw(new Money(7.15));

        var result = new AccountService().CreateAccountReport(account);

        // TODO check all attributes of report
        Assertions.fail("fixme - account service test");
    }

    @Test
    void shouldContainTheSumOfAllDepositTransactions() {
        var owner = new Customer();
        owner.firstName = "Dagobert";
        owner.surName = "Duck";
        var account = new Account(owner);

        account.deposit(new Money(80.7));
        account.withdraw(new Money(7.15));

        var result = new AccountService().CreateAccountReport(account);

        // TODO check all attributes of report
        Assertions.fail("fixme - account service test");
    }

    @Test
    public void shouldCreateAFullAccountReport() {
        var owner = new Customer();
        owner.firstName = "Dagobert";
        owner.surName = "Duck";
        var account = new Account(owner);

        account.deposit(new Money(12.30));
        account.withdraw(new Money(7.15));

        var result = new AccountService().CreateAccountReport(account);

        // TODO check all attributes of report
        Assertions.fail("fixme - account service test");
    }
}

File: CSharp/TestExercisesKatas/TestDoublesExerciseTest/Bank/AccountServiceTest.cs
using FluentAssertions;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;
using TestExercises.Bank;

namespace TestExercises.Bank
{
    class AccountServiceTest
    {
        [Test]
        public void ShouldCreateAccounntReport()
        {
            var owner = new Customer();
            owner.FirstName = "Dagobert";
            owner.SurName = "Duck";
            var account = new Account(owner);

            account.Deposit(new Money(12.30m));

            var result = new AccountService().CreateAccountReport(account);

            // TODO check all attributes of report
            Assert.Fail("fixme - account service test");
        }
    }
}
